cmd/ingester/app/consumer/deadlock_detector.go

```diff
@@ -138,21 +142,25 @@ func (s *deadlockDetector) start() {
 	}
 
 	go func() {
-		s.logger.Debug("Starting global deadlock detector")
-		ticker := time.NewTicker(s.interval)
-		defer ticker.Stop()
-
-		for {
-			select {
-			case <-detector.done:
-				s.logger.Debug("Closing global ticker routine")
-				return
-			case <-ticker.C:
-				if atomic.LoadUint64(detector.msgConsumed) == 0 {
-					s.panicFunc(-1)
-					return // For tests
+		if s.interval == 0 {
```

Comment:
move to head of start() & exit immediately instead?

Reply:
Cannot really do that, because without initialization of "allPartitionsDeadlockDetector" API gets broken - and NullPointerExceptions will start to appear.

I little change this code not to check for interval inside goroutine.